### ISSUE TITLE
Unable to submit iOS app to Apple with cordova-plugin-googlemaps

### ISSUE THREAD

**Issue by grapemix** [mapsplugin/cordova-plugin-googlemaps#1693]
**I'm submitting a ...**  (check one with "x")
[ ] question
[X] any problem or bug report
[ ] feature request

**The plugin version:**  (check one with "x")
[X] **1.4.x**
[] **2.0.0-beta3** 

**If you choose 'problem or bug report', please select OS:**  (check one with "x")
[ ] **Android**
[X] **iOS**

**cordova information:**  (run `$> cordova plugin list`) 

    <plugin name="cordova-plugin-crosswalk-webview" spec="~2.3.0" />
    <plugin name="cordova-plugin-device" spec="~1.1.5" />
    <plugin name="cordova-plugin-dialogs" spec="~1.3.2" />
    <plugin name="cordova-plugin-file" spec="~4.3.2" />
    <plugin name="cordova-plugin-file-transfer" spec="~1.6.2" />
    <plugin name="cordova-plugin-geolocation" spec="~2.4.3">
    </plugin>
    <plugin name="cordova-plugin-googlemaps" spec="~1.4.0">
    </plugin>
    <plugin name="cordova-plugin-openurlext" spec="^1.0.1" />
    <plugin name="cordova-plugin-secure-storage" spec="~2.6.3" />
    <plugin name="cordova-plugin-splashscreen" spec="~4.0.3" />
    <plugin name="cordova-plugin-statusbar" spec="~2.2.2" />
    <plugin name="cordova-plugin-webintent2" spec="~1.0.1" />
    <plugin name="cordova-plugin-whitelist" spec="~1.3.2" />
    <plugin name="cordova-plugin-wkwebview-engine" spec="https://github.com/ionic-team/cordova-plugin-wkwebview-engine" />
    <plugin name="cordova-plugin-x-socialsharing" spec="~5.1.6" />
    <plugin name="cordova.plugins.diagnostic" spec="~3.4.2" />
    <plugin name="ionic-plugin-deeplinks" spec="~1.0.14">
    </plugin>
    <plugin name="ionic-plugin-keyboard" spec="~2.2.1" />
    <plugin name="phonegap-plugin-push" spec="~2.0.0">
    </plugin>
    <plugin name="uk.co.workingedge.phonegap.plugin.launchnavigator" spec="~3.2.1" />

```
insert the output from the command here
```

**Current behavior:**

Build perfectly in phonegap build, but got reject by Apple before manual review
**Expected behavior:**

Build perfectly in phonegap build, accepted by Apple and start manual review process

**Steps to reproduce:**

**Screen capture or video record:**

**Related code, data or error log (please format your code or data):**

```
insert any relevant code, data, or error log here
```

We have been using cordova-plugin-googlemaps + PGB + Application Loader for a while and everything is fine. However, our iOS app was rejected by Apple with the following message. The rejection happened when we used Application Loader to upload our iOS app (before manual rejection process).
 
ERROR ITMS-90596: "Invalid Bundle. The asset catalog at 'Payload/OUR_APP_NAME.app/GoogleMaps.bundle/GMSCoreResources.bundle/Assets.car' can't be read. Try rebuilding the app with a non-beta version of Xcode and submit it again."
 
We have not changed anything specific to google map plugin or iOS related config or PGB related config. Compiling process in PGB is fine. Does anyone have any idea? We have also filed ticket in PGB forum and they suggested us to ask in here. Ty.
----

If your problem is solved, please consider small amount donation to this project.
Appreciate for your kindness.
https://github.com/mapsplugin/cordova-plugin-googlemaps-doc/blob/master/README.md#buy-me-a-beer

**Comment by wf9a5m75** [comment]
`GoogleMaps.bundle/GMSCoreResources.bundle/Assets.car` is Google's file, not this plugin.

I guess you use the old version of `cordova-plugin-googlemaps-sdk`.
I recently updated the version.

At least, you should update the plugin before asking.
https://github.com/mapsplugin/cordova-plugin-googlemaps-sdk

**Comment by grapemix** [comment]
Hi @wf9a5m75,

Thanks for your quick reply. We did used cordova-plugin-googlemaps 1.4.0 before. And we just tested to use cordova-plugin-googlemaps 1.4.5. And we still got the same error.

We never used and heard about the cordova-plugin-googlemaps-sdk plugin. However, PGB said we did install this plugin and it is because cordova-plugin-googlemaps requied it. See
https://github.com/mapsplugin/cordova-plugin-googlemaps/blob/master/plugin.xml: 223

PGB reported cordova-plugin-googlemaps-sdk is installed with mater branch at "0addb07" as git ref. Since cordova-plugin-googlemaps-sdk is automatically installed by cordova-plugin-googlemaps, is there any way we can fix this problem WITHOUT using cordova-plugin-googlemaps 2.x branch?

Ty.

**Comment by wf9a5m75** [comment]
The `cordova-plugin-googlemaps-sdk` is required.
I'm not sure you understand this plugin or not, but the map view is not a HTML element, it is a native view.
In order to generate a native map view, the `cordova-plugin-googlemaps-sdk` is required, and it is developed by Google, not me.
I just add the `plugin.xml` file.

So if Apple rejects your app because of the Google Maps SDK for iOS, 
you need to do with the following things:

- (1) argue to Apple as `it is developed by Google`
- (2) argue to Google as `my app was reject because of the Google Maps SDK for iOS` at https://issuetracker.google.com/bookmark-groups/76561

-------

Anyway, I can not help anything at all.

And the v1.x will be shutdown soon, maybe today or tomorrow.

**Comment by wf9a5m75** [comment]
>ERROR ITMS-90596: "Invalid Bundle. The asset catalog at 'Payload/OUR_APP_NAME.app/GoogleMaps.bundle/GMSCoreResources.bundle/Assets.car' can't be read. Try rebuilding the app with a non-beta version of Xcode and submit it again."

Regarding of the error message, the file `Assets.car` file is exists. 
https://github.com/mapsplugin/cordova-plugin-googlemaps-sdk/blob/master/Maps/Frameworks/GoogleMaps.framework/Resources/GoogleMaps.bundle/GMSCoreResources.bundle/Assets.car

I believe the PGB (PhoneGap Build?) fails to fetch the SDK plugin.

**Comment by grapemix** [comment]
Thanks for your reply. We are very confused too because we have not changed anything related to the problem. 

According to the PhoneGap Build, the log looked fine to us. The log indicated the cordova-plugin-googlemaps-sdk(2.4.0) is fetched and installed without problem. Since the rejection happened before the manual review process, we got no one to "argue" with. And we all know Apple is not really nice to the iOS developer. We didn't have much option here. We are hesitated to switch to 2.X branch because of the beta status, but we may have to give a try.

Finally, thanks again.

**Comment by grapemix** [comment]
After we switched to the 2.0.0-beta3, our android app works perfectly. Our iOS app's map is not responding at all. We do see the map, logo, marker and all button. But all buttons(compass, current location) are not clickable. Dragging, clicking on the map and panning are not responding too. No errors show on building log or on debugging console.

**Comment by wf9a5m75** [comment]
Please share your project files (or invite me to the phone gap build)

**Comment by wf9a5m75** [comment]
ping

**Comment by wf9a5m75** [comment]
ping

**Comment by wf9a5m75** [comment]
Since there is no update for a while, I close this thread.
If you have any updates, you can reopen this thread.

**Comment by grapemix** [comment]
Sorry for the late reply. We have tried almost everything in the past few day, including upgrading to 2.X branch, creating sample project, downgrading and upgrading lots of library, switching to local building environment, in order to reproduce and to isolate the problem. 

However, our last build from PGB mysteriously pass Apple's review process. As we tried so many things and it is hard to tell what solve the problem. We are sorry for your time, but our problem is being solved mysteriously.

Thanks again.
